### PR TITLE
Add dynamic volume provisioning support

### PR DESCRIPTION
**Pull request by aojea**
Dynamic volume provisioning support is nice to have in order to test applications that need persistence.

[Local Path Provisioner](https://github.com/rancher/local-path-provisioner
) provides a way for the Kubernetes users to utilize the local storage in each node ...

It requires a kubernetes version > 1.12

Ref: https://github.com/kubernetes-sigs/kind/issues/118

### CODE REVIEW

**Comment by aojea** [comment]
Please hold on until I can test it locally, I'm having some issues to detect the cluster version

File: pkg/cluster/internal/create/kubeadm-init.go [hunk 1]
```diff
@@ -120,6 +120,40 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
 		return errors.Wrap(err, "failed to add default storage class")
 	}
 
+	// add dynamic volume provisioning
```
Comment (BenTheElder) [inline pkg/cluster/internal/create/kubeadm-init.go]: /hold
I don't think this should be part of kubeadm init, and I think we should consider if we want to bundle this with the node image or not, and if installing this should be configurable.

**Comment by aojea** [comment]
I'll close this PR and wait until https://github.com/kubernetes-sigs/kind/issues/324 that seems aligned with @neolit123 proposal

**Comment by BenTheElder** [comment]
thanks, I actually want to explore this (this weekend?) and bring up at our next meeting, I do think having this either by default or extremely easy to install is important for a lot of usage.

**Pull request closed without merging**